event_management/src/main/java/com/example/event_management/security/AuthorizationFilter.java

```diff
@@ -0,0 +1,149 @@
+package com.example.event_management.security;
+
+import com.example.event_management.exception.UnAuthorisedException;
+import com.example.event_management.model.Session;
+import com.example.event_management.model.Users;
+import com.example.event_management.repository.SessionRepository;
+import com.example.event_management.repository.UsersRepository;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.sql.Timestamp;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Date;
+import java.util.Objects;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Configuration
+public class AuthorizationFilter extends OncePerRequestFilter {
+
+    @Autowired
+    SessionRepository sessionRepository;
+    @Autowired
+    UsersRepository usersRepository;
+
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain) throws IOException, ServletException {
+
+        String requestToken = request.getHeader(HttpHeaders.AUTHORIZATION);
+        String path=request.getRequestURI();
+
+        if(path.equals("/api/login") || path.equals("/api/users")){
+
+            if (Objects.nonNull(requestToken) && !requestToken.isEmpty()) {
+                UsernamePasswordAuthenticationToken authentication = getAuthentication(request,response);
+                SecurityContextHolder.getContext().setAuthentication(authentication);
+            }
+
+            filterChain.doFilter(request, response);
```

Comment:
No need to validate token during login request
```suggestion
            filterChain.doFilter(request, response);
```

Reply:
updated